### PR TITLE
Show the version of dashboard

### LINKED ISSUE #1137: Show Version of Dashboard Component
### Feature request

I would like to see what version of the dashboard I have installed on my Kubernetes cluster via `tkn`.

### Use case

For reporting issues to dashboard, it would be helpful for users to easily see what version of the dashboard is deployed. It's also a convenient way for users to know what features are available in a particular version of the dashboard. 

### UI Example

```
$ tkn version

Client version: 0.11.0
Pipeline version: v0.11.0
Triggers version: v0.6.0
Dashboard version: v0.6.0
```

### PR DESCRIPTION
**Pull request by rudeigerc**
# Changes

Closes #1137 

Show the version of the dashboard when executing `tkn version`.

```
$ tkn version
Client version: dev
Pipeline version: v0.15.2
Triggers version: v0.7.0
Dashboard version: v0.8.2
```
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

**Comment by rudeigerc** [comment]
It seems like there is something wrong with the configuration of testing?

**Comment by vdemeester** [comment]
> It seems like there is something wrong with the configuration of testing?

Yeah the go coverage has a tiny problem, see https://github.com/tektoncd/plumbing/pull/543 that will fix that :angel: sorry about that :pray:

**Comment by rudeigerc** [comment]
@vdemeester That's all right! 😄 Since the PR has been merged I believe the tests will work this time.
/test pull-tekton-cli-go-coverage

**Comment by nikhil-thomas** [comment]
@rudeigerc this patch looks great. :+1: 
this is a great improvement to tkn ux.

One minor concern this patch brought to light is that, the `Triggers Version` and `Dashboard Version` will be `Unknown` even when they are not installed on a cluster.

I understand and appreciate that this patch is consistent with the current `tkn version` ux.

Let us get this patch merged.
I shall create a new issue so that we can discuss the `version: unknown` part more.

cc @danielhelfand, @vdemeester @piyush-garg @pradeepitm12

**Comment by vdemeester** [comment]
> @rudeigerc this patch looks great. +1
> this is a great improvement to tkn ux.
> 
> One minor concern this patch brought to light is that, the `Triggers Version` and `Dashboard Version` will be `Unknown` even when they are not installed on a cluster.
> 
> I understand and appreciate that this patch is consistent with the current `tkn version` ux.
> 
> Let us get this patch merged.
> I shall create a new issue so that we can discuss the `version: unknown` part more.
> 
> cc @danielhelfand, @vdemeester @piyush-garg @pradeepitm12

Thx @nikhil-thomas @rudeigerc. I do agree with @nikhil-thomas. What I would like is for all components (except `pipeline`) to only be displayed when available. Which would mean, in case we don't find triggers, we don't display in in `tkn version`. Same would apply for dashboard. Ideally I would like to have this in 0.12.0, but we can do a follow-up on this :wink:

**Comment by rudeigerc** [comment]
Thanks for the reviews! @nikhil-thomas @vdemeester 
I have seen the discussion in #1143. Is there something that needs to be changed in this patch, or handling them in another patch would be better?

**Comment by nikhil-thomas** [comment]
> Thanks for the reviews! @nikhil-thomas @vdemeester
> I have seen the discussion in #1143. Is there something that needs to be changed in this patch, or handling them in another patch would be better?

@rudeigerc This patch is good-to-merge, in my opinion.

The issue #1143 can be handled in a different patch.
From the issue thread (https://github.com/tektoncd/cli/issues/1143#issuecomment-680146033), @Divyansh42 has picked it up. :heart:

**Review by vdemeester — APPROVED** [review]
/meow
/cc @piyush-garg @pradeepitm12

File: pkg/version/version.go [hunk 1]
```diff
@@ -130,3 +132,33 @@ func findTriggersVersion(deployments []v1.Deployment) string {
...
+func GetDashboardVersion(c *cli.Clients) (string, error) {
+	deploymentsList, err := getDeployments(c, dashboardControllerSelector, oldDashboardControllerSelector)
+
+	if err != nil {
+		return "", err
+	}
+
+	version := findDashboardVersion(deploymentsList.Items)
+	if version == "" {
+		return "", fmt.Errorf("Error getting the tekton dashboard deployment version. Version is unknown")
```
Comment (danielhelfand) [inline pkg/version/version.go:146]: Nit: Lowercase this error message

```suggestion
		return "", fmt.Errorf("error getting the tekton dashboard deployment version: version is unknown")
```

Comment (rudeigerc) [reply pkg/version/version.go:146]: Thanks for the review! Actually I copied the error handling from above (pipeline and trigger). 😂
Should they also be fixed?

Comment (danielhelfand) [reply pkg/version/version.go:146]: Yes, not your fault. There have definitely been misses about enforcing go error practices in reviews. You are more than welcome to update the other errors, but not a blocker to update them.

**Comment by rudeigerc** [comment]
@danielhelfand Sorry for my mistake. I probably referred to a wrong version.

I discover that in the deployment definition of v0.6.0, there is no version specified in the labels of the pod template.
Maybe getting the version from metadata directly using `deployment.GetObjectMeta().GetLabels()` would be better.

```shell
Name:                   tekton-dashboard
Namespace:              tekton-pipelines
CreationTimestamp:      Wed, 26 Aug 2020 22:41:41 +0800
Labels:                 app=tekton-dashboard
                        version=v0.6.0
Annotations:            deployment.kubernetes.io/revision: 1
Selector:               app=tekton-dashboard
Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           app=tekton-dashboard
  Service Account:  tekton-dashboard
  ...
```

File: pkg/version/version.go (hunk 1 above)
Comment (rudeigerc) [reply pkg/version/version.go:146]: I get it. Thank you very much! 😄

**Review by piyush-garg — APPROVED** [review]

**Comment by chmouel** [comment]
/lgtm
/approve

Thanks 👍🏼🔥

**Pull request merged**